### ISSUE TITLE
Irrigation System

### ISSUE THREAD

**Issue by miticolars** [snowdd1/homebridge-knx#180]
Hi, thank you for your work and for this plugin. Is it possible to configure an Irrigation System with this plugin? I've a KNX relay module that drives some valves of my irrigation system and I would to control it with homebridge. Thank you.

**Comment by snowdd1** [comment]
That might be possible, I think there are services for irrigation and valves. But I don't have any so I cannot test that myself.

**Comment by miticolars** [comment]
Yes, there is the IrrigationSystem service but is not clear how can I set the knx_config.json to use my knx relay module (obviously I know all the knx address needed). Or if needed to develop some code. Thank you

**Comment by marcobastianon** [comment]
hi @miticolars i'm plaing to crate some irigation zone using ( like you ) knx switch.. i've create this configuration:

        {
            "DeviceName": "Goccia Aiuola",
            "Services": [
                {
                    "ServiceType": "IrrigationSystem",
                    "ServiceName": "Goccia Aiuola",
                    "Characteristics": [
                        {
                            "Type": "Active",
                            "Set": ["1/1/68"],
                            "Listen": ["1/2/68"]
                        },
                        {"Type": "ProgramMode"},
                        {
                            "Type": "InUse",
                            "Listen": ["1/2/68"]
                        },
                        {"Type": "RemainingDuration"}
                    ]
                }
            ],
        },

i've status of irrigation correctly, but if i click on the icon nothing start.

**Comment by miticolars** [comment]
Hi Marco, thank you. I think that IrrigationSystem must include some valves (zones), but I don’t know the correct syntax. I’m working on it. Paolo

**Comment by miticolars** [comment]
@marcobastianon I tried changing the Service from IrrigationSystem to Valve and it works, but I would to use the IrrigationSystem Service to drive my valves. Let me know if you find some way to do it. Thanks!

**Comment by marcobastianon** [comment]
maybe we need to write this service?

**Comment by miticolars** [comment]
@marcobastianon try this:

`{
	"DeviceName": "Goccia aiuola",
	"Services": [
		{                                    
			"ServiceType": "Valve",          
			"ServiceName": "Goccia aiuola",           
			"Characteristics": [             
				{                            
					"Type": "Active",        
					"Set": [                 
						"1/1/68"              
					],                       
					"Listen": [              
						"1/1/68"              
					],                       
					"DPT": "DPT1"            
				},                           
				{                            
					"Type": "InUse",         
					"Listen": [              
						"1/1/68"             
					],                       
					"DPT": "DPT1"            
				},                           
				{                            
					"Type": "ValveType"      
				},                           
				{                            
					"Type": "SetDuration"    
				},                           
				{                             
					"Type": "RemainingDuration" 
				}                               
			]
		}                                       
	]
}`

The icon is not the correct one (I would want to see the sprinkler icon and not the faucet) and the timer doesn't work.
Bye

**Comment by marcobastianon** [comment]
how can we set { "Type": "ValveType" } to 1?

**Comment by miticolars** [comment]
> how can we set { "Type": "ValveType" } to 1?

I don't know how to do it, if you find the solution just let me know

**Comment by snowdd1** [comment]
> > how can we set { "Type": "ValveType" } to 1?
> 
> I don't know how to do it, if you find the solution just let me know

You could use a group address that answers with 1 - in that case no programming required. Of course you need to have such an address.

**Comment by miticolars** [comment]
> > > how can we set { "Type": "ValveType" } to 1?
> > 
> > 
> > I don't know how to do it, if you find the solution just let me know
> 
> You could use a group address that answers with 1 - in that case no programming required. Of course you need to have such an address.

yeah, that's a solution, i try to find an address that can be useful in this way. Thanks!

**Comment by marcobastianon** [comment]
@snowdd1 thx!!
![image](https://user-images.githubusercontent.com/23148549/93329254-33802e80-f81d-11ea-9f9a-bbe6a7bfe87c.png)

works!! super fantastik!

now i've only the temperature issue :) realy lovely!!

**Comment by miticolars** [comment]
> @snowdd1 thx!!
> ![image](https://user-images.githubusercontent.com/23148549/93329254-33802e80-f81d-11ea-9f9a-bbe6a7bfe87c.png)
> 
> works!! super fantastik!
> 
> i need only to work with default time.. start all time @00.00 infinite
> 
> now i've only the temperature issue :) realy lovely!!

@marcobastianon can you use the timer to run the sprinkler (remaining time, etc...)?

**Comment by marcobastianon** [comment]
no, i've added "SetDuration" and i can see on bus 300s (5min) but nothing decrease and time is still 00.00 (infinite) i think we need to decrees using handler or from knx logic module. (i preff from handler)

**Comment by miticolars** [comment]
> > > how can we set { "Type": "ValveType" } to 1?
> > 
> > 
> > I don't know how to do it, if you find the solution just let me know
> 
> You could use a group address that answers with 1 - in that case no programming required. Of course you need to have such an address.

I don't  have an address that answer always 1. I have an address that answer always 0, but the Reverse attribute doesn't work (if I understand what this atrribute do).

**Comment by marcobastianon** [comment]
you can send 1 @ new group address, with ets maybe.. the valveType
1=sprinkler

**Comment by ctschach** [comment]
I have created an custom handler which also supports the timer when using the eve app.

You can head over to 

https://github.com/ctschach/homebridge-knx

and test out this version. I'm about to do some final clean ups and then submit a pull request.

**Comment by miticolars** [comment]
> I have created an custom handler which also supports the timer when using the eve app.
> 
> You can head over to
> 
> https://github.com/ctschach/homebridge-knx
> 
> and test out this version. I'm about to do some final clean ups and then submit a pull request.

Thank you for your time and your job! 
I tested your version and I'm able to set the correct sprinkler icon and the timer.

Great!
